pkg/machine/define/vmfile.go

```diff
@@ -48,6 +49,22 @@ func (m *VMFile) Read() ([]byte, error) {
 	return os.ReadFile(m.GetPath())
 }
 
+// Read the first n bytes of a given file and return in []bytes
+func (m *VMFile) ReadMagicNumber(n int) ([]byte, error) {
+	f, err := os.Open(m.GetPath())
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	b := make([]byte, n)
+	n, err = io.ReadFull(f, b)
+	if err != nil && err != io.ErrUnexpectedEOF && err != io.EOF {
+		return b[:n], err
+	} else {
+		return b[:n], nil
+	}
```

Comment:
nit: the else not required since you return above in the if block
But not worth worth to repush over that.